File: drivers/gpu/drm/nouveau/nvkm/engine/fifo/gb202.c
/* SPDX-License-Identifier: MIT
 *
 * Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
 */
#include "priv.h"
#include "cgrp.h"
#include "chan.h"
#include "runl.h"

u32
gb202_chan_doorbell_handle(struct nvkm_chan *chan)
{
	return BIT(30) | (chan->cgrp->runl->id << 16) | chan->id;
}

File: drivers/gpu/nova-core/fsp/hal/gb202.rs
// SPDX-License-Identifier: GPL-2.0
// SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.

use kernel::io::Io;

use crate::{
    driver::Bar0,
    fsp::hal::FspHal,
    regs, //
};

struct Gb202;

impl FspHal for Gb202 {
    fn fsp_boot_status(&self, bar: Bar0<'_>) -> u32 {
        bar.read(regs::gb202::NV_THERM_I2CS_SCRATCH_FSP_BOOT_COMPLETE)
            .fsp_boot_complete()
            .into()
    }

    fn cot_version(&self) -> u16 {
        2
    }
}

const GB202: Gb202 = Gb202;
pub(super) const GB202_HAL: &dyn FspHal = &GB202;
